internal/test_cases/invalid_command_test_case.go

```diff
@@ -0,0 +1,59 @@
+package test_cases
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/codecrafters-io/shell-tester/internal/assertions"
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+type InvalidCommandTestCase struct {
+	Command string
+}
+
+func (t *InvalidCommandTestCase) RunAndTestReflection(asserter *logged_shell_asserter.LoggedShellAsserter, shell *shell_executable.ShellExecutable, logger *logger.Logger) error {
+	testCase := CommandReflectionTestCase{
+		Command:             t.Command,
+		SkipPromptAssertion: true,
+	}
+	if err := testCase.Run(asserter, shell, logger, true); err != nil {
+		return err
+	}
+
+	asserter.AddAssertion(assertions.SingleLineAssertion{
+		ExpectedOutput:   t.getExpectedOutput(),
+		FallbackPatterns: t.getFallbackPatterns(),
+	})
+
+	if err := asserter.AssertWithoutPrompt(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (t *InvalidCommandTestCase) RunAndTestResponse(asserter *logged_shell_asserter.LoggedShellAsserter, shell *shell_executable.ShellExecutable, logger *logger.Logger) error {
```

Comment:
@andy1li can you explain how `RunAndTestResponse` is different from `RunAndTestReflection`? Both do seem to test the response and reflection, no? (I see `ExpectedOutput` used in both + the second one uses `CommandResponseTestCase` which internally uses `CommandReflectionTestCase`).

Reply:
@rohitpaulk TBH, I'm not 100% sure why we needed `CommandReflectionTestCase` and `CommandResponseTestCase` in the first place. Just didn't want to break the current code.

This is the comment explaining the use of `CommandReflectionTestCase` in stage 2:

>
	// We are seperating this out because we don't want to assert
	// The prompt at the end
	testCase := test_cases.CommandReflectionTestCase{

https://github.com/codecrafters-io/shell-tester/blob/main/internal/stage2.go#L25-L30

Reply:
@andy1li the naming still doesn't make sense - `RunAndTestResponse` does test reflection, and `RunAndTestReflection` does test the command response. Wanna try digging into the code a bit more and coming up with cleaner names? Feel free to restructure things a bit if the existing names can be improved.

Reply:
Ah, got it! REPL is not expected to be implemented at this stage. So `CommandReflectionTestCase` is used to skip testing the prompt $ at the next iteration.

Reply:
@rohitpaulk How about `Run` and `RunButSkipNextPrompt`?

Reply:
@andy1li yep, works! Small tweak I'd suggest: `Run` and `RunWithoutNextPromptAssertion`. This is a bit more explicit and more consistent with other usages (we have a `RunWithout...` somewhere else too)